kratos/processes/fast_transfer_between_model_parts_process.cpp

```diff
@@ -108,25 +113,27 @@ void FastTransferBetweenModelPartsProcess::TransferWithFlags()
 {
     // Creating a buffer for parallel vector fill
     const int num_threads = OpenMPUtils::GetNumThreads();
-    std::vector<NodesArrayType> nodes_buffer(num_threads);
-    std::vector<ElementsArrayType> elements_buffer(num_threads);
-    std::vector<ConditionsArrayType> conditions_buffer(num_threads);
+    std::vector<NodesArrayType> nodes_buffer_vector(num_threads);
+    std::vector<ElementsArrayType> elements_buffer_vector(num_threads);
+    std::vector<ConditionsArrayType> conditions_buffer_vector(num_threads);
+    std::vector<MasterSlaveConstraintArrayType> constraints_buffer_vector(num_threads);
 
     // Auxiliar sizes
     const int num_nodes = static_cast<int>(mrOriginModelPart.Nodes().size());
     const int num_elements = static_cast<int>(mrOriginModelPart.Elements().size());
     const int num_conditions = static_cast<int>(mrOriginModelPart.Conditions().size());
+    const int num_constraints = static_cast<int>(mrOriginModelPart.Conditions().size());
 
     #pragma omp parallel
     {
-        const int id = OpenMPUtils::ThisThread();
+        const int thread_id = OpenMPUtils::ThisThread();
```

Comment:
do you really need the thread_id ? the best would be not to use this, unless it is strictly needed

Reply:
Yes, in order to create a buffer for each thread. This is already done, just adding constraints